srcs/Client.cpp

```diff
@@ -342,31 +305,45 @@ void Client::setConfig(ConfigGroup &group)
 	}
 }
 
+void Client::makeBasicHeader()
+{
+	StartLineRes &start_line = this->_response.getStartLine();
+
+	start_line.protocol = "HTTP/1.1";
+	this->_response.insertToHeader("Server", "418-IAmATeapot");
+	this->_response.insertToHeader("Date", getCurrentTime());
+}
+
 /**
  * Client::makeMsg
  *
  */
 void Client::makeMsg()
 {
-	StartLineRes &res_start_line = this->_response.getStartLine();
-	StartLineReq &req_start_line = this->_request.getStartLine();
-	
-	std::cout << req_start_line << std::endl;
+	StartLineReq &start_line = this->_request.getStartLine();
 
-	res_start_line.protocol = "HTTP/1.1";
-	this->_response.insertToHeader("Server", "418-IAmATeapot");
-	this->_response.insertToHeader("Date", getCurrentTime());
-	this->_response.insertToHeader("Allow", makeMethodList(this->_config_location->method));
-	/**
-	 * TODO: config에 없는 method가 들어왔을 때에 대한 response 처리
-	 * (405 Method Not Allowed)
-	 */
-	switch (req_start_line.method)
+	std::cout << start_line << std::endl;
+
+	if (!this->_config_location->auth.empty())
+	{
+		std::vector<std::string> temp = ft_split(this->_request.getHeaderValue("Authorization"), ' ');
+		if (temp[1].empty())
+			throw 401;
+		if (!temp[1].compare(this->_config_location->auth))
+			throw 403;
+	}
```

Comment:
```suggestion
	if (!this->_config_location->auth.empty())
	{
		std::string auth_value = this->_request.getHeaderValue("Authorization");
		if (auth_value.empty())
			throw 401;
		std::vector<std::string> temp = ft_split(auth_value, ' ');
		if (temp.length != 2 || temp[0].compare("Basic") || temp[1].compare(this->_config_location->auth))
			throw 403;
	}
```

- Request Header에 Authorization이 없는 경우
- Authorization 방식이 Basic이 아닌경우
- string compare 함수 실행시 ! 제거

등 문제가 있을 거 같아 위와 같이 구현해야 될 것 같습니다

Reply:
저도 한 번 다시 훑어봤고 이 부분만 수정해주시면 될 것 같아요~ 👌